packages/www/components/Dashboard/Logger/index.tsx

```diff
@@ -44,52 +62,83 @@ function orchestratorName({
   return !matches?.length ? address : matches[1];
 }
 
-function handleEvent(evt: events.Any, userIsAdmin: boolean): LogData[] {
-  switch (evt.type) {
-    case "stream_state":
-      const state = evt.state.active ? "active" : "inactive";
-      const region = evt.region || "unknown";
-      return [
-        {
-          key: evt.id,
-          timestamp: evt.timestamp,
-          level: "info",
-          text: `Stream is ${state} in region "${region}"`,
-        },
-      ];
-    case "transcode":
-      if (evt.success && !userIsAdmin) {
-        // non-admin users should only see fatal errors.
-        return [];
-      }
-      return evt.attempts
-        .filter((a) => a.error)
-        .map((a, idx) => ({
-          key: `${evt.id}-${idx}`,
-          timestamp: evt.timestamp,
-          level: "error",
-          text: `Transcode error from ${orchestratorName(a)} for segment ${
-            evt.segment.seqNo
-          }: ${a.error}`,
-        }));
-    case "webhook_event":
-      if (!evt.event.startsWith("multistream.")) {
-        console.error("unknown event:", evt.event);
-        break;
+function createEventHandler() {
+  const lastOrchestrator = useRef<string>();
+  const failedSegments = useRef(new Set<number>());
+
+  return useCallback(
+    function handleEvent(evt: events.Any, userIsAdmin: boolean): LogData[] {
+      switch (evt.type) {
+        case "stream_state":
+          const state = evt.state.active ? "active" : "inactive";
+          const region = evt.region || "unknown";
+          return [infoLog(evt, `Stream is ${state} in region "${region}"`)];
+        case "transcode":
+          const logs: LogData[] = [];
+          const seqNo = evt.segment.seqNo;
+
+          // non-admin users should only see fatal errors.
+          if (!evt.success || userIsAdmin) {
+            const errLogs = evt.attempts
+              .filter((a) => a.error)
+              .map((a, idx) => {
+                const orch = orchestratorName(a);
+                const msg = `Transcode error from ${orch} for segment ${seqNo}: ${a.error}`;
+                return errorLog(evt, msg, `error-${idx}`);
```

Comment:
On this error log, is the orchestrator error sent on the dashboard? Do we need this kind of extensive information about the orchestrator failure to transcode/deliver the segment to be provided to the user? Or is it in any case the most generic piece of information we have about the error?

Reply:
We only provide the error to user in case all attempts to transcode the segment have failed (`!evt.success`). That likely means that they are ingesting some bad media to us, and should definitely see the error that comes out of the transcoding process. This was one of the main goals we had with stream health, so that users can diagnose any issues they might have with their stream that they were previously unable to identify before. One example of those logs is if you send the video with an unsupported pixel format, and you'll get that error on the dashboard through this log.

On the worst case, it might be the case that we are having some issues in our infrastructure and the transcoding is failing because of that. That still means that the users' stream is going to fail, and I think it's better for us to show some error to them anyway, even if it is internal. If they want high-level information, they can also check only the healthchecks anyway and will see the transcoding/realtime/etc "boolean" health conditions about the stream, which should be easier for the less technical users. 

This is all pretty new and not iterated enough anyway, so we're likely changing strategies in the future on how to display some of this information for the user. For now, that is the mindset tho, high-level status on the healthchecks and low-level info in the logs.